File: faunadb-java/src/test/java/com/faunadb/client/MetricsResponseSpec.java
package com.faunadb.client;

import com.faunadb.client.types.MetricsResponse;
import com.faunadb.client.types.Value;
import org.junit.Before;
import org.junit.Test;

import java.util.Map;
import java.util.Optional;

import static org.junit.Assert.assertEquals;

public class MetricsResponseSpec {

    Map<MetricsResponse.Metrics, String> metricsMap;

    @Before
    public void setUp() {
        metricsMap = Map.of(
                MetricsResponse.Metrics.QUERY_BYTES_IN, "35",
                MetricsResponse.Metrics.QUERY_BYTES_OUT, "75"
        );
    }

    @Test
    public void shouldReturnCorrectValueObject() {
        Value value = new Value.LongV(25L);
        MetricsResponse metricsResponse = MetricsResponse.of(value, metricsMap);

        assertEquals(new Value.LongV(25L), metricsResponse.getValue());
    }

    @Test
    public void shouldReturnEmptyIfMetricNotExists() {
        MetricsResponse metricsResponse = MetricsResponse.of(null, metricsMap);

        assertEquals(Optional.empty(), metricsResponse.getMetric(MetricsResponse.Metrics.BYTE_READ_OPS));
    }

    @Test
    public void shouldReturnCorrectMetricValue() {
        MetricsResponse metricsResponse = MetricsResponse.of(null, metricsMap);

        assertEquals(Optional.of("35"), metricsResponse.getMetric(MetricsResponse.Metrics.QUERY_BYTES_IN));
        assertEquals(Optional.of("75"), metricsResponse.getMetric(MetricsResponse.Metrics.QUERY_BYTES_OUT));
    }
}

File: faunadb-scala/src/test/scala/faunadb/MetricsResponseSpec.scala
package faunadb

import faunadb.values.Metrics.Metrics
import faunadb.values.{LongV, Metrics, MetricsResponse}
import org.scalatest.BeforeAndAfterAll
import org.scalatest.flatspec.AnyFlatSpec
import org.scalatest.matchers.should.Matchers

class MetricsResponseSpec extends AnyFlatSpec with Matchers with BeforeAndAfterAll {

  val metricsMap: Map[Metrics, String] = Map(
    Metrics.QueryBytesIn -> "35",
    Metrics.QueryBytesOut -> "75"
  )

  it should "return correct value object" in {
    val value = LongV(25L)
    val metricsResponse = MetricsResponse(value, metricsMap)

    metricsResponse.value shouldBe LongV(25L)
  }

  it should "return empty if metric not exists" in {
    val metricsResponse = MetricsResponse(null, metricsMap)

    metricsResponse.getMetric(Metrics.ByteReadOps) shouldBe None
  }

  it should "return correct metric value" in {
    val metricsResponse = MetricsResponse(null, metricsMap)

    metricsResponse.getMetric(Metrics.QueryBytesIn) shouldBe Some("35")
    metricsResponse.getMetric(Metrics.QueryBytesOut) shouldBe Some("75")
  }
}
